pyccel/ast/low_level_macros.py

```diff
@@ -0,0 +1,61 @@
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/devel/LICENSE for full license details.      #
+#------------------------------------------------------------------------------------------#
+"""
+Module to handle low-level language agnostic types such as macros.
+"""
+
+from .basic import PyccelAstNode
+
+__all__ = ('MacroDefinition',)
+class MacroDefinition(PyccelAstNode):
+    """
+    A class for defining a macro in a file.
+
+    A class for defining a macro in a file.
+
+    Parameters
+    ----------
+    macro_name : str
+        The name of the macro.
+    obj : Any
+        The object that will define the macro.
+    suffix : str, optional
+        A suffix that may be added to the object.
+    """
+    _attribute_nodes = ()
+    __slots__ = ('_macro_name', '_obj', '_suffix')
+
+    def __init__(self, macro_name, obj, suffix = None):
```

Comment:
Should't errors be raised if the arguments are not of the right type?

Reply:
630bf364ef47ac6fac0fc5a66d4a56a0c0bc589b

Reply:
I used assertions instead of raises as these errors are for devs and assertions can be deactivated